service_grpc_flight/src/lib.rs

```diff
@@ -634,11 +638,11 @@ where
 
 /// Extracts an encoded Protobuf message from a [`FlightDescriptor`],
 /// as used in FlightSQL.
-fn msg_from_descriptor(flight_descriptor: FlightDescriptor) -> Result<Any> {
+fn cmd_from_descriptor(flight_descriptor: FlightDescriptor) -> Result<FlightSQLCommand> {
     match flight_descriptor.r#type() {
         DescriptorType::Cmd => {
-            let msg: Any = Message::decode(flight_descriptor.cmd).context(DeserializationSnafu)?;
-            Ok(msg)
+            let cmd = FlightSQLCommand::try_decode(flight_descriptor.cmd)?;
+            Ok(cmd)
```

Comment:
```suggestion
            FlightSQLCommand::try_decode(flight_descriptor.cmd)
```
Perhaps, not sure about the error types